File: src/main/java/fi/quanfoxes/Status.java
package fi.quanfoxes;

public class Status {
    public static final Status OK = new Status("OK", false);

    private String description;
    private boolean problematic;

    public Status(String description, boolean problematic) {
        this.description = description;
        this.problematic = problematic;
    }

    public String getDescription() {
        return description;
    }

    public boolean isProblematic() {
        return problematic;
    }

    public static Status error(String format, Object... args) {
        return new Status(String.format(format, args), true);
    }

    public static Status error(String description) {
        return new Status(description, true);
    }
}

File: Zigzag/Status.cs
using System.Globalization;

public class Status
{
	public static readonly Status OK = new Status("OK", false);

	public string Description { get; }

	public bool IsProblematic { get; private set; }

	private Status(string description, bool problematic)
	{
		Description = description;
		IsProblematic = problematic;
	}

	public static Status Error(string format, params object[] arguments)
	{
		return new Status(string.Format(CultureInfo.InvariantCulture, format, arguments), true);
	}

	public static Status Error(string description)
	{
		return new Status(description, true);
	}
}